pkg/operator/v1helpers/helpers.go

```diff
@@ -57,11 +57,13 @@ func SetOperatorCondition(conditions *[]operatorv1.OperatorCondition, newConditi
 		return
 	}
 
-	if existingCondition.Status != newCondition.Status {
-		existingCondition.Status = newCondition.Status
+	if existingCondition.Status != newCondition.Status ||
```

Comment:
change in `status` or `reason` should be considered a transition, correct?

Reply:
No, it is the 
```
// Last time the condition transitioned from one status to another.
```
I have seen some conditions that have a `lastUpdateTime` also (like `DeploymentCondition`).

Reply:
thanks @sanchezl, so it applies to the condition as a whole, the last time the condition changed state, irrespective of the reason. I think it makes sense.